pageserver/src/tenant/timeline.rs

```diff
@@ -1473,16 +1459,44 @@ impl Timeline {
     }
 
     pub(super) fn launch_wal_receiver(
-        &self,
+        self: &Arc<Self>,
         ctx: &RequestContext,
-        broker_client: BrokerClientChannel,
-    ) -> anyhow::Result<()> {
+        broker_client: &'static BrokerClientChannel,
+    ) {
         info!(
             "launching WAL receiver for timeline {} of tenant {}",
             self.timeline_id, self.tenant_id
         );
-        self.walreceiver.start(ctx, broker_client)?;
-        Ok(())
+
+        let tenant_conf_guard = self.tenant_conf.read().unwrap();
+        let wal_connect_timeout = tenant_conf_guard
+            .walreceiver_connect_timeout
+            .unwrap_or(self.conf.default_tenant_conf.walreceiver_connect_timeout);
+        let lagging_wal_timeout = tenant_conf_guard
+            .lagging_wal_timeout
+            .unwrap_or(self.conf.default_tenant_conf.lagging_wal_timeout);
+        let max_lsn_wal_lag = tenant_conf_guard
+            .max_lsn_wal_lag
+            .unwrap_or(self.conf.default_tenant_conf.max_lsn_wal_lag);
+        drop(tenant_conf_guard);
+
+        let mut guard = self.walreceiver.lock().unwrap();
+        assert!(
```

Comment:
just to make sure I follow, we moved to assert instead of the runtime checks that were in wal_reciver::start ?
```rust
if self.started.load(atomic::Ordering::Acquire) {
            anyhow::bail!("Wal receiver is already started");
        }

        let timeline = self.timeline_ref.upgrade().with_context(|| {
            format!("walreceiver start on a dropped timeline {}", self.timeline)
        })?;
```

Reply:
We made `Timeline::launch_wal_receiver` responsible for ensuring it.
And it asserts this by checking that the `Timeline::walreceiver` is `None` before launching it and setting it to `Some()`.